Makefile

```diff
@@ -386,7 +395,7 @@ BTREE_SYS = $(OBJDIR)/$(SRCDIR)/btree.o           \
 # Note each test bin/unit/<x> also depends on obj/unit/<x>.o, as
 # defined above using unit_test_self_dependency.
 #
-$(BINDIR)/$(UNITDIR)/misc_test: $(UTIL_SYS)
+$(BINDIR)/$(UNITDIR)/misc_test: $(UTIL_SYS) $(COMMON_UNIT_TESTOBJ)
```

Comment:
I thought about including this new `.o` in `COMMON_TESTOBJ` symbol, but that would mean we are mixing up shared `.o's` from diff sub-dirs. 

The way it's done is a bit more work to keep adding this `COMMON_UNIT_TESTOBJ` dependency in several lines, but it's at least clear, and cleaner, to specify exactly which `unit_test-dot-oh` depends on which common object.